### DISCUSSION TITLE
How to hoist `eslint-plugin-XXX`?

### DISCUSSION THREAD

**Discussion by loynoir** [yarnpkg/berry#4446] [Q&A] [answered]
What's the yarn berry equivalent of pnpm `public-hoist-pattern[]=*eslint*`?

### Pnpm's public-hoist-pattern docs

https://pnpm.io/npmrc#public-hoist-pattern

### What works in pnpm
Given
- package `plugin-bundle` have dependency `eslint-plugin-foo`, `eslint-plugin-bar`, ...
- package `A` depends on package `plugin-bundle` 

A pnpm user with `public-hoist-pattern[]=*eslint*`, `eslint` works correctly in package `A`.

### Question
I found docs on field `nohoist`, but said it's yarn v1.
How can I hoist `*eslint*` to top using yarn berry?

`.yarnrc.yml`

```txt
nodeLinker: pnpm

yarnPath: .yarn/releases/yarn-3.2.0.cjs
```

Or, in another word, given
- If have package `A` with tens of `eslint-plugin-XXX` dependency, which I think too ugly
- I concat all `eslint-plugin-XXX` packages into one `plugin-bundle` package

How can I let `A` have one `plugin-bundle` dependency, instead of ugly tens of `eslint-plugin-XXX`, and makes `eslint` happy?

### Else
PS: I set `public-hoist-pattern[]=*eslint*` within `~/.npmrc`, and have `nodeLinker: pnpm`, but pnpm's `public-hoist-pattern` is not working.

**Comment by merceyz** [comment, marked as answer]
There is no hoisting for the `pnpm` linker and Yarn doesn't let you [control the hoisting](https://yarnpkg.com/configuration/yarnrc#nmHoistingLimits) at such a low level for the `node-modules` linker.
For ESLint setups I suggest using [@rushstack/eslint-patch](https://github.com/microsoft/rushstack/tree/d77a6a73cbb0b645e01ab1a21de8a43c515e3476/eslint/eslint-patch) to correct its resolution.
